### ISSUE TITLE
Modin==0.6.0 requires Dask even if you just want to use Ray

### ISSUE THREAD

**Issue by CMobley7** [modin-project/modin#785]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 18.04
- **Modin installed from (source or binary)**:  binary
- **Modin version**: 0.6.0
- **Python version**: 3.7.3

### Describe the problem
After running `pip install modin[ray]` and then using 
```
import os
os.environ["MODIN_ENGINE"] = "ray"  # Modin will use Ray
import modin.pandas as pd
```
I get 
```
import modin.pandas as pd
  File "/usr/local/lib/python3.6/dist-packages/modin/pandas/__init__.py", line 80, in <module>
    from .concat import concat
  File "/usr/local/lib/python3.6/dist-packages/modin/pandas/concat.py", line 2, in <module>
    from .dataframe import DataFrame
  File "/usr/local/lib/python3.6/dist-packages/modin/pandas/dataframe.py", line 21, in <module>
    from .utils import from_pandas, to_pandas, _inherit_docstrings
  File "/usr/local/lib/python3.6/dist-packages/modin/pandas/utils.py", line 1, in <module>
    from ..data_management.factories import BaseFactory
  File "/usr/local/lib/python3.6/dist-packages/modin/data_management/factories.py", line 209, in <module>
    class PandasOnDaskFactory(BaseFactory):
  File "/usr/local/lib/python3.6/dist-packages/modin/data_management/factories.py", line 211, in PandasOnDaskFactory
    from modin.engines.dask.pandas_on_dask_futures.io import PandasOnDaskIO
  File "/usr/local/lib/python3.6/dist-packages/modin/engines/dask/pandas_on_dask_futures/io.py", line 7, in <module>
    from modin.engines.dask.pandas_on_dask_futures.frame.data import PandasOnDaskFrame
  File "/usr/local/lib/python3.6/dist-packages/modin/engines/dask/pandas_on_dask_futures/frame/data.py", line 1, in <module>
    from distributed.client import _get_global_client
ModuleNotFoundError: No module named 'distributed'
```
It looks like even though I set the backend to `Ray`, `Modin` still expects to be able to access `Dask` for some reason. Any ideas?

**Comment by devin-petersohn** [comment]
Thanks @CMobley7, we have caught this one earlier and already have a fix in master (#780). #784 is a followup fix for the files in the experimental API.

We are planning a release for 0.6.1 later this week, but you can also install from the master: https://modin.readthedocs.io/en/latest/installation.html#installing-from-the-github-master-branch

**Comment by devin-petersohn** [comment]
0.6.1 was released, please reopen if the issue persists.
